soa-derive-internal/src/lib.rs

```diff
@@ -34,14 +35,34 @@ pub fn soa_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
     generated.append_all(index::derive(&input));
     generated.append_all(iter::derive(&input));
     generated.append_all(derive_trait(&input));
+
+    {
+        generated.append_all(
+            generic::derive_slice(&input)
+        );
+        generated.append_all(
+            generic::derive_slice_mut(&input)
+        );
+        generated.append_all(
+            generic::derive_vec(&input)
+        );
+    }
     generated.into()
 }
 
 use crate::input::Input;
 use quote::quote;
+
+#[allow(unused)]
 fn derive_trait(input: &Input) -> TokenStream {
     let name = &input.name;
     let vec_name = names::vec_name(name);
+    let slice_name = names::slice_name(name);
+    let slice_mut_name = names::slice_mut_name(name);
+    let ref_name = names::ref_name(name);
+    let ref_mut_name = names::ref_mut_name(name);
+    let ptr_name = names::ptr_name(name);
+    let ptr_mut_name = names::ptr_mut_name(name);
```

Comment:
```suggestion
fn derive_trait(input: &Input) -> TokenStream {
    let name = &input.name;
    let vec_name = names::vec_name(name);
```

this does not seems used anywhere